### ISSUE TITLE
Update docs in preparation for 0.6.0

### ISSUE THREAD

**Issue by dotsdl** [datreant/datreant.core#13]
Many new features and changes have been made since the docs were originally written (which was for MDSynthesis back in January). These need to be updated with fresh eyes in light of these changes, and then vetted by others.

I'll be working on this over the weekend. Anyone interested in helping to make them better after a first draft? @cing, @sseyler? I'll ping this issue when I've finished an alpha version.

**Comment by sseyler** [comment]
@dotsdl I'd be more than happy to use my fresh eyes to vet your updates to the docs.

**Comment by dotsdl** [comment]
@sseyler thanks! This will help a lot.

**Comment by dotsdl** [comment]
Whew...I think the docs for this release are complete. They have been almost completely rewritten given all the work that's been done on the library. Please [have a look at the docs as they are now](http://datreant.readthedocs.org/en/develop/) and write here if there's anything confusing, missing, or terrible.

If @orbeckst, @kain88-de, @cing , or @sseyler could take a gander before I ship the release tomorrow morning (PDT) that would be swell!

**Comment by orbeckst** [comment]
Overall: very good, ready to roll. The logical ordering is good, in particular, introducing groups as persistent bundles.

Minor comments:

- spell check (... unambibuously ;-) but also others that a spell checker would find)
- http://datreant.readthedocs.org/en/develop/tags-categories.html : highlight with a Note or Warning that None cannot be use as a category value
- I like the split into "prose" docs and API reference. However, I would make more liberal use of links to the API ref, e.g. use :class:`Treant` markup or when you talk about other methods. Right now there are few links to classes/methods/functions. It is also a bit confusing if the same method is either clickable and black or non-linked and red (eg discover on http://datreant.readthedocs.org/en/develop/bundles.html#gathering-treants-from-the-filesystem ). For consistency I'd always use the sphinx roles; you can use :meth:`~datreant.core.discover` if you only want the name itself.
- I think that the boolean selections of the tag interface are a horrible hack http://datreant.readthedocs.org/en/develop/bundles.html#using-tag-expressions-to-select-members ...
- If you already do fuzzy matching why not also regex?
- groupby is very nice!
- map is also great!
- How is a missing member announced? Raises exception?  http://datreant.readthedocs.org/en/develop/groups.html#groups-find-their-members-when-they-go-missing
- The "Frequently" Asked Questions are good.





On 21 Mar, 2016, at 22:09, David Dotson wrote:

> Whew...I think the docs for this release are complete. They have been almost completely rewritten given all the work that's been done on the library. Please have a look at the docs as they are now and write here if there's anything confusing, missing, or terrible.

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by dotsdl** [comment]
I think I've addressed most of these, but I'll respond to a few that I didn't just resolve:

1. I don't think it's necessary to link every mention of ``Treant`` (I don't think you're saying that), but I did make sure I linked anything that was alredy put into ``literal`` backticks that made sense. Thanks for the tip on the "~" shortenings!
2. I think the boolean selections are really useful, though I'll admit it feels a bit strange to use. We might as well keep it since `__getitem__` for tags might as well do something, but there are probably other ways we could implement similar tag querying. Open to alternatives.
3. We can add ``Tags.regex``. Happy to. We could also add ``Tags.glob`` without any fuss, too, for those that prefer that (me :D). I'll make it an issue for the next milestone.
